### PR TITLE
feat: Pass a single struct with all the request parameters

### PR DESCRIPTION
**Pull request by oblique**

### CODE REVIEW

File: src/interface/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+
+impl From<U256> for Nat {
+    fn from(value: U256) -> Self {
+        let mut bytes = [0u8; 32];
+        value.0.to_little_endian(&mut bytes);
+        Nat(BigUint::from_bytes_le(&bytes))
+    }
+}
+
+#[derive(Debug, Clone, CandidType, Deserialize)]
```
Comment (zvolin) [inline src/interface/src/lib.rs]: wondering if it'd be good to have also `PartialEq, Eq`

**Review by zvolin — APPROVED** [review]
lgtm

File: src/ethereum-canister/src/lib.rs [hunk 2]
```diff
@@ -1,28 +1,28 @@
...
 use ic_cdk::{init, query, update};
+use interface::{Address, Erc20OwnerOfRequest, Erc721OwnerOfRequest, SetupRequest, U256};
 
 mod erc20;
 mod erc721;
 mod helios;
 mod random;
 mod utils;
 
-use crate::utils::ToNat;
```
Comment (zvolin) [inline src/ethereum-canister/src/lib.rs:10]: should we remove ToNat now?

Comment (oblique) [reply src/ethereum-canister/src/lib.rs:10]: we can, but idk if we will need it later

**Pull request merged**